components/scream/src/mct_coupling/tests/scorpio_tests.cpp

```diff
@@ -107,30 +218,153 @@ TEST_CASE("scorpio_interface_output", "") {
+  get_variable(outfilename,"y","y-direction",1,vec_y, PIO_REAL,"y-real_in");
+  get_variable(outfilename,"z","z-direction",1,vec_z, PIO_REAL,"z-real_in");
+  get_variable(outfilename,"data_1d","test value for 1d field",1,vec_x, PIO_REAL,"x-real_in");
+  get_variable(outfilename,"data_2d","test value for 2d field",2,vec_xy, PIO_REAL,"xy-real_in");
+  get_variable(outfilename,"data_3d","test value for 3d field",3,vec_xyz, PIO_REAL,"xyz-real_in");
+  get_variable(outfilename,"index_1d","test value for 1d field",1,vec_x, PIO_INT,"x-int_in");
+  get_variable(outfilename,"index_2d","test value for 2d field",2,vec_xy, PIO_INT,"xy-int_in");
+  get_variable(outfilename,"index_3d","test value for 3d field",3,vec_xyz, PIO_INT,"xyz-int_in");
+  /* 
+   * Setup PIO configuation for reading data.  This is usually a 3 step process:
+   * 1. Determine the number of indices, start and stop location for this rank.
+   * 2. Create a vector of the indices based on the info from step 1. 
+   *    We need to do this as a second step since we don't know the length of this vector ahead of time.
+   * 3. Assign the global indices to the dof vector.
+   * We already did all of this setting things up for writing output, so here we just reuse the same dof vectors.
+   */
+  set_dof(outfilename,"time",0,0);
+  set_dof(outfilename,"x",dof_x[0],x_dof.data());
+  set_dof(outfilename,"y",dof_y[0],y_dof.data());
+  set_dof(outfilename,"z",dof_z[0],z_dof.data());
+  set_dof(outfilename,"index_1d",dof_test_1d[0],test1d_dof.data());
+  set_dof(outfilename,"data_1d",dof_test_1d[0],test1d_dof.data());
+  set_dof(outfilename,"index_2d",dof_test_2d[0],test2d_dof.data());
+  set_dof(outfilename,"data_2d",dof_test_2d[0],test2d_dof.data());
+  set_dof(outfilename,"index_3d",dof_test_3d[0],test3d_dof.data());
+  set_dof(outfilename,"data_3d",dof_test_3d[0],test3d_dof.data());
+  // Now that the variables have been gathered and the dof set for each one, set the iodesc decomposition from PIO.
+  set_decomp(outfilename);
+  /* Read input data for x, y and z, and compare with expected value.  Report mismatches as an error. */
+  grid_read_data_array(outfilename,"x",dof_x[0],ekat::util::data(x_data)+xstart);
+  grid_read_data_array(outfilename,"y",dof_y[0],ekat::util::data(y_data)+ystart);
+  grid_read_data_array(outfilename,"z",dof_z[0],ekat::util::data(z_data)+zstart);
+  nerr = 0;
+  for (Int ii=xstart;ii<=xstop;ii++) {
+    if (x_data[ii] != x_i(ii,x_data.size())) { ++nerr;}
+  }
+  REQUIRE(nerr==0);
+  nerr = 0;
+  for (Int jj=ystart;jj<=ystop;jj++) {
+    if (y_data[jj] != y_i(jj,y_data.size())) { ++nerr;}
+  }
+  REQUIRE(nerr==0);
+  nerr = 0;
+  for (Int kk=zstart;kk<=zstop;kk++) {
+    if (z_data[kk] != z_i(kk,z_data.size())) { ++nerr; std::printf("   - (%d) %f vs. %f\n",kk,z_data[kk],z_i(kk,z_data.size()));}
+  }
+  REQUIRE(nerr==0);
+  /* Now read and compare the field data for each of the timestep */
+  nerr = 0;
```

Comment:
You don't use nerr between here and the next `nerr=0;`. Did you forge to pass it to some function call? Or perhaps you don't need to set it here?

Reply:
I think you are right, originally I thought that maybe I would get a read out for all the errors and I wanted to keep a tally over each of the variables.  But seeing as how the moment REQUIRE throws an error everything shuts down, reinitializing `nerr` is probably not needed.  I've removed all of these.